extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -322,17 +326,30 @@ export class AmpStoryPage extends AMP.BaseElement {
     this.maybeConvertCtaLayerToPageOutlink_();
   }
 
-  /** @private */
-  maybeSetPreviewDuration_() {
-    if (this.storeService_.get(StateProperty.PREVIEW_STATE)) {
-      const videos = this.getAllVideos_();
-
-      const autoAdvanceAttr =
-        videos.length > 0
-          ? VIDEO_PREVIEW_AUTO_ADVANCE_DURATION
-          : DEFAULT_PREVIEW_AUTO_ADVANCE_DURATION;
-
-      this.element.setAttribute('auto-advance-after', autoAdvanceAttr);
+  /**
+   * Configures the page to auto advance using preview-specific durations.
+   * @private
+   */
+  setupAutoAdvanceForPreview_() {
+    const firstVideo = this.getFirstAmpVideo_();
+    const autoAdvanceDuration = firstVideo
+      ? VIDEO_PREVIEW_AUTO_ADVANCE_DURATION
+      : DEFAULT_PREVIEW_AUTO_ADVANCE_DURATION;
+    this.element.setAttribute('auto-advance-after', autoAdvanceDuration);
+
+    if (firstVideo) {
+      whenUpgradedToCustomElement(firstVideo)
+        .then(() => firstVideo.getImpl())
+        .then((videoImpl) => {
+          const loopVideoIfTooShort = (duration) => {
```

Comment:
Now you can prevent creating this function here, and just running this code inside the promise `.then`